autosklearn/pipeline/components/regression/mlp.py

```diff
@@ -0,0 +1,239 @@
+        self.tol = tol
+        self.solver = solver
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        self.epsilon = epsilon
+        self.beta_1 = beta_1
+
+        self.random_state = random_state
+        self.verbose = verbose
+        self.estimator = None
+
+    @staticmethod
+    def get_max_iter():
+        return 512
+
+    def get_current_iter(self):
+        return self.estimator.n_iter_
+
+    def iterative_fit(self, X, y, n_iter=2, refit=False):
+        """
+        Set n_iter=2 for the same reason as for SGD
+        """
+        from sklearn.neural_network import MLPRegressor
+        import sklearn.preprocessing
+        n_iter = max(n_iter, 2)
+
+        if refit:
+            self.estimator = None
+            self.scaler = None
+
+        if self.estimator is None:
+            self._fully_fit = False
+
+            self.max_iter = int(self.max_iter)
+            self.hidden_layer_depth = int(self.hidden_layer_depth)
+            self.num_nodes_per_layer = int(self.num_nodes_per_layer)
+            self.hidden_layer_sizes = tuple(self.num_nodes_per_layer
+                                            for i in range(self.hidden_layer_depth))
+            self.activation = str(self.activation)
+            self.alpha = float(self.alpha)
+            self.learning_rate_init = float(self.learning_rate_init)
+            self.early_stopping = str(self.early_stopping)
+            self.early_stopping_ = str(self.early_stopping)
+            if self.early_stopping == "train":
+                self.validation_fraction = 0.0
+                self.tol = float(self.tol)
+                self.n_iter_no_change = int(self.n_iter_no_change)
+                self.early_stopping_val = False
+            elif self.early_stopping == "valid":
+                self.validation_fraction = float(self.validation_fraction)
+                self.tol = float(self.tol)
+                self.n_iter_no_change = int(self.n_iter_no_change)
+                self.early_stopping_val = True
+            else:
+                raise ValueError("Set early stopping to unknown value %s" % self.early_stopping)
+            # elif self.early_stopping == "off":
+            #     self.validation_fraction = 0
+            #     self.tol = 10000
+            #     self.n_iter_no_change = self.max_iter
+            #     self.early_stopping_val = False
+
+            self.solver = self.solver
+
+            try:
+                self.batch_size = int(self.batch_size)
+            except ValueError:
+                self.batch_size = str(self.batch_size)
+
+            self.shuffle = check_for_bool(self.shuffle)
+            self.beta_1 = float(self.beta_1)
+            self.beta_2 = float(self.beta_2)
+            self.epsilon = float(self.epsilon)
+            self.beta_1 = float(self.beta_1)
+
+            self.max_iter = int(self.max_iter)
```

Comment:
This argument isn't casted for classification -> I'm not sure which one is correct.

Reply:
it is, but for regression is was casted twice -> done